src/docs/04-java-client/18-workflow-replay-shadowing.md

```diff
@@ -0,0 +1,157 @@
+
+#### Shadow Mode
+
+- Normal: Shadowing will complete after all workflows matches WorkflowQuery (after sampling) have been replayed or when exit condition is met.
+- Continuous: A new round of shadowing will be started after all workflows matches WorkflowQuery have been replayed. There will be a 5 min wait period between each round, and currently this wait period is not configurable. Shadowing will complete only when ExitCondition is met. ExitCondition must be specified when using this mode.
+
+#### Shadow Concurrency
+
+- Concurrency: workflow replay concurrency. If not specified, will be default to 1. For local shadowing, an error will be returned if a value higher than 1 is specified.
+
+### Local Shadowing Test
+
+Local shadowing test is similar to the replay test. First create a workflow shadower with optional shadow and replay options, then register the workflow that need to be shadowed. Finally, call the `Run` method to start the shadowing. The method will return if shadowing has finished or any non-deterministic error is found. 
+
+Here's a simple example. The example is also available [here](https://github.com/uber/cadence-java-samples/blob/master/src/test/java/com/uber/cadence/samples/hello/HelloWorkflowShadowingTest.java).
+
+```java
+public void testShadowing() throws Throwable {
+  IWorkflowService service = new WorkflowServiceTChannel(ClientOptions.defaultInstance());
+
+  ShadowingOptions options = ShadowingOptions
+          .newBuilder()
+          .setDomain(DOMAIN)
+          .setShadowMode(Mode.Normal)
+          .setWorkflowTypes(Lists.newArrayList("GreetingWorkflow::getGreeting"))
+          .setWorkflowStatuses(Lists.newArrayList(WorkflowStatus.OPEN, WorkflowStatus.CLOSED))
+          .setExitCondition(new ExitCondition().setExpirationIntervalInSeconds(60))
+          .build();
+  WorkflowShadower shadower = new WorkflowShadower(service, options, TASK_LIST);
+  shadower.registerWorkflowImplementationTypes(HelloActivity.GreetingWorkflowImpl.class);
+
+  shadower.run();
+}
+```
+
+### Shadowing Worker
+
+NOTE: 
+- **All shadow workflows are running in one Cadence system domain, and right now, every user domain can only have one shadow workflow at a time.**
+- **The Cadence server used for scanning and getting workflow history will also be the Cadence server for running your shadow workflow.** Currently, there's no way to specify different Cadence servers for hosting the shadowing workflow and scanning/fetching workflow.
+
+Your worker can also be configured to run in shadow mode to run shadow tests as a workflow. This is useful if there's a number of workflows need to be replayed. Using a workflow can make sure the shadowing won't accidentally fail in the middle and the replay load can be distributed by deploying more shadow mode workers. It can also be incorporated into your deployment process to make sure there's no failed replay checks before deploying your change to production workers.
```

Comment:
This is useful if there's a number of workflows `that` need to be replayed

Reply:
done